File: BrickBreakerADT/src/com/carlospinan/brickbreakerandroid/utils/Rectangle.java
package com.carlospinan.brickbreakerandroid.utils;

/**
 * Created by mac on 14/07/13.
 */
public class Rectangle {

	public float left, top, right, bottom;

	public Rectangle(float left, float top, float right, float bottom) {
		this.left = left;
		this.top = top;
		this.right = right;
		this.bottom = bottom;
	}

	public boolean intersect(Rectangle r) {
		return left <= r.right && right >= r.left && top <= r.bottom
				&& bottom >= r.top;
	}

}

File: BrickBreaker/app/src/main/java/com/carlospinan/brickbreaker/game/models/Rectangle.kt
package com.carlospinan.brickbreaker.game.models

/**
 * @author Carlos Piñan
 */
class Rectangle(
    val left: Float,
    val top: Float,
    val right: Float,
    val bottom: Float
) {

    companion object {
        fun make(left: Float, top: Float, right: Float, bottom: Float): Rectangle {
            return make(left, top, right, bottom)
        }
    }

    fun intersect(left: Float, top: Float, right: Float, bottom: Float): Boolean {
        return intersect(make(left, top, right, bottom))
    }

    fun intersect(rectangle: Rectangle): Boolean {
        return left <= rectangle.right &&
                right >= rectangle.left &&
                top <= rectangle.bottom &&
                bottom >= rectangle.top
    }

}